src/components/input/_macro.njk

```diff
@@ -71,21 +71,26 @@
         {% endif %}
 
         {% if params.prefix is defined and params.prefix or params.suffix is defined and params.suffix %}
-            {% if params.prefix is defined and params.prefix %}
-                {% set prefixClass = " ons-input-type--prefix" %}
-            {% endif %}
+            {% if (params.prefix.id is defined and params.prefix.id) or (params.suffix.id is defined and params.suffix.id) %}
```

Comment:
```suggestion
            {% if (params.prefix is defined and params.prefix and params.prefix.id is defined and params.prefix.id) or (params.suffix is defined and params.suffix and params.suffix.id is defined and params.suffix.id) %}
```
This line is causing an error in jinja2 unless you make this change. 
It also means you can get rid of the if above it but if you do that, not setting the id will still render inputs without no suffix/preffix. If you leave it in there will be nothing rendered if the id isn't set instead. Im not sure which functionality you're going for

Reply:
Yeah I ideally we should prevent both the input and prefix rendering if they don't set the required attributes.  Should make it easier to see when something is missing I think